resources/lib/globals.py

```diff
@@ -215,6 +215,17 @@ def init_globals(self, argv):
         self.TIME_TRACE_ENABLED = self.ADDON.getSettingBool('enable_timing')
         self.IPC_OVER_HTTP = self.ADDON.getSettingBool('enable_ipc_over_http')
 
+        import resources.lib.database.db_local as db_local
+        self.LOCAL_DB = db_local.NFLocalDatabase()
+        import resources.lib.database.db_shared as db_shared
+        import resources.lib.database.db_utils as db_utils
+        # TODO: xml settings to specify a custom path
+        #  need to study how better apply to client/service
```

Comment:
For the client, it's not a problem as it's checked on every launch.

For the service, I'd use the settings_monitor and 'onSettingChange' compare the current DB path to the settings DB path.

**If changed:**
If the file already exists in the new path, use it (it's most probably a file moved from an another client)
Else if it doesn't exists, move (more copy and delete) the current db file to the new location
change self.SHARED_DB to the new value

Reply:
We also need to ensure the path is avaliable, ie the SMB folder is not offline, to not create a file again in addons_data/plugin.video.netflix

Reply:
I can say that this is what i thought, you added some more details
but now i have two other things that i have to change, so then i'll think about it
we will see next week